xdsl/interactive/get_rewrites_and_passes.py

```diff
@@ -0,0 +1,98 @@
+from typing import NamedTuple
+
+from xdsl.dialects import builtin
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir import MLContext
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.tools.command_line_tool import get_all_dialects, get_all_passes
+from xdsl.transforms import individual_rewrite
+from xdsl.transforms.mlir_opt import MLIROptPass
+
+ALL_PASSES = tuple(sorted((p_name, p()) for (p_name, p) in get_all_passes().items()))
+"""Contains the list of xDSL passes."""
+
+
+class IndividualRewrite(NamedTuple):
+    """
+    Type alias for a possible rewrite, described by an operation and pattern name.
+    """
+
+    operation: str
+    pattern: str
+
+
+class IndexedIndividualRewrite(NamedTuple):
+    """
+    Type alias for a specific rewrite pattern, additionally consisting of its operation index.
+    """
+
+    operation_index: int
+    rewrite: IndividualRewrite
+
+
+def get_all_possible_rewrites(
+    patterns: tuple[IndividualRewrite, ...],
+    op: ModuleOp,
+    rewrite_by_name: dict[str, dict[str, RewritePattern]],
+) -> tuple[IndexedIndividualRewrite, ...]:
+    """
+    Function that takes a sequence of IndividualRewrite Patterns and a ModuleOp, and
+    returns the possible rewrites.
+    """
+    old_module = op.clone()
+    num_ops = len(list(old_module.walk()))
+
+    current_module = old_module.clone()
+
+    res: tuple[IndexedIndividualRewrite, ...] = ()
+
+    for op_idx in range(num_ops):
+        matched_op = list(current_module.walk())[op_idx]
+        if matched_op.name not in individual_rewrite.REWRITE_BY_NAMES:
```

Comment:
this is the global variable, not the one passed into the function

Reply:
I don't understand what the requested change is.

Reply:
```suggestion
        if matched_op.name not in rewrite_by_name:
```